### ISSUE TITLE
Bottlerocket worker node support

### ISSUE THREAD

**Issue by cnmcavoy** [kubernetes-sigs/cluster-api-provider-aws#4708]
/kind feature

**Describe the solution you'd like**
As a user, I would like to be able spawn bottlerocket worker nodes in CAPA-owned clusters. I can manually specify the AMI ID of Bottlerocket from the AMI catalog, but this is insufficient to have the workers join a cluster, the workers are created but never "join" and add the node to the cluster, so pod workloads can not schedule. This affects both EKS and non-EKS clusters.

**Anything else you would like to add:**
What seems to be missing is a way to configure the Bottlerocket user data. Bottlerocket is immutable at runtime, so configuration must be done before the node is created. Bottlerocket takes a TOML-formatted configuration format that must be supplied as user-data, which is where configuration such as the api-server endpoint, certificate data, cluster name, etc are. 

CAPA seems to be unaware Bottlerocket has a custom configuration format and needs to be enhanced to be aware of it, in order to supply the correct settings for a node to join a cluster.

https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2009 seems to be related, but only describes simplifying the AMI selection. However, manually specifying the AMI ID of Bottlerocket directly is not sufficient to join a cluster at present.

**Comment by praveenadini** [comment]
@cnmcavoy I was able to provision bottlerocket nodes and join it to the cluster by passing a custom bootstrap secret with userdata information to the MachinePool object but I see  disk pressure issues with BottleRocket nodes as the /dev/xvdb comes with 20GB of data storage (used for containers, etc) which I couldn't figure out a way to expand using AWSManagedMachinePool -> LaunchTemplate and because of this some of the pods are ending up in Eviction, ContainerStatusUnknown states.

**Comment by kahirokunn** [comment]
It would be nice if the bottlerocket could be quickly specified by amiType, like AL2.

**Comment by kahirokunn** [comment]
Keep

**Comment by kahirokunn** [comment]
keep

**Comment by josemrs** [comment]
Honestly, I don't understand why this is open/not possible yet. It's also confusing.

I can see here https://cluster-api-aws.sigs.k8s.io/crd/#infrastructure.cluster.x-k8s.io/v1beta2.ManagedMachineAMIType it should be already possible to specify Bottlerocket as amitype.

However, when I tried I get an error that only types XYZ... (The ones from 1beta1) Are supported and I'm using 1beta2 .

As I said, confusing.

**Comment by richardcase** [comment]
@josemrs - my understanding that some work will still need to be done to make sure the userdata is correct for bottlerocket.

**Comment by richardcase** [comment]
> Honestly, I don't understand why this is open/not possible yet. It's also confusing.

@josemrs  - would you be interested in helping out adding support for this?

**Comment by josemrs** [comment]
> > Honestly, I don't understand why this is open/not possible yet. It's also confusing.
> 
> [@josemrs](https://github.com/josemrs) - would you be interested in helping out adding support for this?

Sorry, I was just pointing that the situation is confusing, perhaps due to some misunderstanding of mine.

**Comment by richardcase** [comment]
It would be great to have this support.

**Comment by yuminn-k** [comment]
I’d like to help with this. I checked current `main` and do not see a competing PR. Bottlerocket AMI types are already mapped for managed node groups, but the bootstrap generation paths I found are still cloud-init / bootstrap.sh / nodeadm-oriented. The nodeadm proposal also lists OSs with different bootstrapping mechanisms like Bottlerocket as a non-goal.

Before I start coding, could maintainers confirm the preferred shape?

- introduce a dedicated Bottlerocket bootstrap config/type that renders TOML user-data;
- add a Bottlerocket mode to an existing bootstrap config;
- or scope this initially to EKS managed node groups vs self-managed Machines/MachinePools?

Once direction is confirmed, I can start with a small design/POC for TOML user-data generation plus unit tests.
